File: 2011-final-value-of-variable-after-performing-operations.java
import java.util.*;

class FinalValueOfVariableAfterPerformingOperationsSolution {

    public int finalValueAfterOperations(String[] operations) {
        return Arrays.asList(operations).stream().map(op -> op.contains("+") ? 1 : -1).reduce(0, Integer::sum);
    }
    public int finalValueAfterOperations2(String[] operations) {
        int res = 0;
        for (String op : operations) {
            int temp = op.contains("+") ? 1 : -1;
            res += temp;
        }
        return res;
    }

    public static void main(String[] args) {
        FinalValueOfVariableAfterPerformingOperationsSolution sol = new FinalValueOfVariableAfterPerformingOperationsSolution();
        System.out.println(
                "Output: " + sol.finalValueAfterOperations(new String[] { "--X", "X++", "X++" }) + ", expected: 1");
        System.out.println(
                "Output: " + sol.finalValueAfterOperations(new String[] { "++X", "++X", "X++" }) + ", expected: 3");
        System.out.println("Output: " + sol.finalValueAfterOperations(new String[] { "X++", "++X", "--X", "X--" })
                + ", expected: 0");
    }

}

File: 2011-final-value-of-variable-after-performing-operations.py
class Solution:
    def finalValueAfterOperations(self, operations: list[str]) -> int:
        '''
        We just need to go through each element and add 1 if we see '+' and deduct 1 if we see '-'
        '''
        res = 0
        for op in operations:
            if '+' in op:
                res += 1
            else:
                res -= 1
        
        return res


sol = Solution()
print(
    f'output: {sol.finalValueAfterOperations(["--X", "X++", "X++"])}, expected: 1')
print(
    f'output: {sol.finalValueAfterOperations(["++X", "++X", "X++"])}, expected: 3')
print(
    f'output: {sol.finalValueAfterOperations(["X++", "++X", "--X", "X--"])}, expected: 0')
